pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4214 @@
+        instance_store = self.get_instance_store(namespace)
+
+        if not class_store.exists(ClassName):
+            raise CIMError(
+                CIM_ERR_NOT_FOUND,
+                _format("Class {0!A} in namespace {1!A} not in CIM repository. "
+                        "Nothing deleted.", ClassName, namespace))
+
+        # create list of subclass names and append the target class
+        classnames = self._get_subclass_names(ClassName, class_store, True)
+        classnames.append(ClassName)
+
+        # Delete all instances in this class and subclasses and delete
+        # this class and subclasses
+        for clname in classnames:
+            sub_clns = self._get_subclass_list_for_enums(ClassName, namespace,
+                                                         class_store)
+
+            inst_paths = [inst.path for inst in instance_store.iter_values()
+                          if inst.path.classname in sub_clns]
+
+            for ipath in inst_paths:
+                instance_store.delete(ipath)
+
+            class_store.delete(clname)
+
+    ##########################################################
+    #
+    #              Faked Qualifier methods
+    #
+    ###########################################################
+
+    ####################################################################
+    #
+    #   EnumerateQualifiers server request
+    #
+    ####################################################################
+
+    def EnumerateQualifiers(self, namespace):
+        """
+        Imlements a  WBEM server responder for
+        :meth:`~pywbem.WBEMConnection.EnumerateQualifiers`
+
+        Enumerates the qualifier declarations in the local CIM repository of
+        namespace.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            Name of the namespace containind the qualifier declarations to be
+            returned (case independent).
+
+            Leading and trailing slash characters will be stripped. The lexical
+            case will be preserved.
+
+        Returns:
+          A list of :class:`~pywbem.CIMQualifierDeclaration` objects that are
+          representations of the enumerated qualifier declarations.
+
+        Raises:
+          CIMError: CIM_ERROR_INVALID_NAMESPACE
+        """
+
+        # Validate namespace and get qualifier_store for this namespace
+        qualifier_store = self.get_qualifier_store(namespace)
+
+        # pylint: disable=unnecessary-comprehension
+        qualifiers = [q for q in qualifier_store.iter_values()]
+
+        return qualifiers
+
+    ####################################################################
+    #
+    #   GetGetQualifier server request
+    #
+    ####################################################################
+
+    def GetQualifier(self, namespace, QualifierName):
+        """
+        Implements a WBEM server responder for
+        :meth:`pywbem.WBEMConnection.GetQualifier`.
```

Comment:
Missing tilde before `pywbem.WBEM...`.

Actually, if I think about it, since we have so many different methods with the operation names, it wouzld not hurt to consistently remove the tilde on methods with the operation names, so be explicit in the docs which ones we reference.

Reply:
TODO DISCUSSION

Reply:
We agreed to omit the ~ from things like :meth:`~pywbem.WBEMConnection.GetQualifier`

Reply:
done